feat(k8s): add tuto cilium (#4156)
scaleway/docs-content: tutorials/enabling-encryption-in-kapsule-with-cilium/index.mdx

## Creating a `CiliumNodeConfig` resource for encryption

The `CiliumNodeConfig` resource defines encryption settings for Cilium. It enables **WireGuard encryption** across all nodes in your Kapsule cluster.

1. Prepare the Resource Definition by saving the following content to a file named `cilium-encryption.yaml`:
    ```
    apiVersion: cilium.io/v2
    kind: CiliumNodeConfig
    metadata:
    namespace: kube-system
    name: enable-encryption
    spec:
    nodeSelector:
        matchLabels: {}
    defaults:
        enable-wireguard: "true"
        enable-wireguard-userspace-fallback: "false"
        wireguard-persistent-keepalive: "0s"
        encrypt-node: "false"
    ```
    <Message type="note">
    * `matchLabels: {}` applies encryption to **all nodes** in the cluster.  
    * Adjust `nodeSelector` if you want to target specific nodes (e.g., nodes labeled `k8s.scaleway.com/managed: "true"`).
    </Message>
2. Deploy the `CiliumNodeConfig` resource to your cluster using `kubectl`:
    ```
    kubectl apply -f cilium-encryption.yaml
    ```

> ## Restarting the Cilium DaemonSet**

Reviewer: ```suggestion
## Restarting the Cilium DaemonSet
```

After creating the `CiliumNodeConfig`, you must restart Cilium to apply these encryption settings.

1. Rollout Restart Cilium using `kubectl`
    ```
    kubectl rollout restart daemonset cilium -n kube-system
    ```

2. Wait for rollout completion. You can check the status of the rollout using the following command:
    ```
    kubectl rollout status daemonset cilium -n kube-system
    ```

3. Verify the pod status using the following `kubectl` command:
    ```
    kubectl get pods -n kube-system -l k8s-app=cilium
    ```
    All Cilium pods should eventually show as **Running** and **Ready**.

> ## Validating Encryption

Reviewer: ```suggestion
## Validating encryption
```

In this step, you will deploy test applications along with a `tcpdump` DaemonSet to observe network traffic before and after enabling encryption.

### 3.1 Deploying test applications and `tcpdump`

Below is an example YAML manifest that deploys:

* An **nginx** Deployment and a corresponding **Service**.
* A **curl** Deployment that continuously makes requests to **nginx**.
* A **tcpdump** DaemonSet to capture traffic on each node’s `kapsule0` interface.

> 1. Save the following as `test-and-tcpdump.yaml`:

Reviewer: ```suggestion
1. Save the following template in a `test-and-tcpdump.yaml` file:
```

        terminationGracePeriodSeconds: 1
        containers:
            - name: tcpdump
            image: alpine:latest
            securityContext:
                privileged: true
            command: ["sh"]
            args:
                - -c
                - |
                apk add --no-cache tcpdump
                # Verify the Kapsule interface
                if ! ip link show kapsule0; then
                    echo "kapsule0 not found"
                    exit 1
                fi
                # Capture traffic on kapsule0 for ports 80 or 51871 (WireGuard)
                tcpdump -Anevi kapsule0 -T vxlan 2>&1 | grep -E '(([0-9]+\.){3}[0-9]+\.(80|51871))'
            resources:
                limits:
                cpu: 100m
                memory: 100Mi
    ```

2. Apply the manifest using `kubectl`:
    ```
    kubectl apply -f test-and-tcpdump.yaml
    ```

### Observing traffic before encryption

If you have **not yet** applied the `CiliumNodeConfig` and restarted Cilium, you should see traffic on **port 80** in the logs of the `tcpdump` pods:

```
kubectl logs -n default daemonset/tcpdump -f
```

(Select any pod if multiple logs are shown.)

> ### Checking encryption status in Cilium**

Reviewer: ```suggestion
### Checking encryption status in Cilium
```
